### PR TITLE
add `is_letter_file_final` & fix CodeClimate Coverage & Smells

### PR DESCRIPTION
**Pull request by rindibudiaramdhan**
## Changes
- add `is_letter_file_final` to `vaccine_requests table`
- move `App\VaccineRequest` to `App\Models\Vaccine\VaccineRequest`
- validate vaccine products payload
- fix function `rules` has a cognitive complexity
- fix `rules` has a cognitive complexity
- `scopeFilter` has 32 linesof code
- fix `purposes` response
- fix `bug` testing
- method `scopeFilter` has 27 lines of code (exceeds 25 allowed)

## Evidence
title: add `is_letter_file_final` & fix CodeClimate Coverage & Smells
project: Pikobar Logistik
participants: @rindibudiaramdhan @rizkyrmsyah @ayocodingit 

cc: @jabardigitalservice/jds-backend

### CODE REVIEW

**Comment by rindibudiaramdhan** [comment]
Sorry, this should be separate to 2 PR, but I made it to one.

File: app/Http/Requests/StoreVaccineRequest.php [hunk 1]
```diff
@@ -23,19 +39,52 @@ public function authorize()
...
-            'primary_phone_number' => 'required|numeric',
-            'logistic_request' => 'required|json',
-            'letter_file' => 'required|mimes:jpeg,jpg,png,pdf|max:10240',
-            'applicant_file' => 'mimes:jpeg,jpg,png,pdf|max:10240',
-            'application_letter_number' => 'required|string'
+            'valid' => $valid,
+            'fails' => $fails,
         ];
     }
 }
```
Comment (ayocodingit) [inline app/Http/Requests/StoreVaccineRequest.php:90]: Wah logic yang panjang yah  😁😅

Comment (rindibudiaramdhan) [reply app/Http/Requests/StoreVaccineRequest.php:90]: yup! cara baca untuk json value yang baru saja saya dapatkan dari laracast, @ayocodingit

Comment (rindibudiaramdhan) [reply app/Http/Requests/StoreVaccineRequest.php:90]: https://laravel.com/docs/7.x/validation#using-closures

**Review by rizkyrmsyah — APPROVED** [review]

File: app/Http/Requests/StoreVaccineRequest.php (hunk 1 above)
Comment (ayocodingit) [reply app/Http/Requests/StoreVaccineRequest.php:90]: > yup! cara baca untuk json value yang baru saja saya dapatkan dari laracast, @ayocodingit 

Dari frontend nya data yang dikirim kan nya berupa json kah?

Comment (rindibudiaramdhan) [reply app/Http/Requests/StoreVaccineRequest.php:90]: json mang @ayocodingit

**Pull request merged**